OutOfSchool/OutOfSchool.BusinessLogic/Services/Memento/MementoService.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using OutOfSchool.BusinessLogic.Services.Memento.Interfaces;
+
+namespace OutOfSchool.BusinessLogic.Services.Memento;
+
+/// <summary>
+/// Implements the IMementoService{T} interface with CRUD functionality to store an entity of type T in a cache.
+/// </summary>
+/// <typeparam name="T">T is the entity type that should be stored in the cache.</typeparam>
+public class MementoService<T> : IMementoService<T>
+{
+    private readonly ICrudCacheService crudCacheService;
```

Comment:
There are all CRUD operations in ICrudCacheService - Create and Update (UpsertValueAsync), Read (GetValueAsync) and Delete (RemoveAsync).


Add my discussion with [VadymLevkovskyi](https://github.com/VadymLevkovskyi) about Code duplication in CasheService class:

> > (Code duplication) This method looks like a copy of part of the `GetOrAdd()` implementation. If it is so and to avoid duplication could it be called in `GetOrAdd()` instead of existing code? If it is not a duplication - then why? For me it seems to do same thing
> 
> I agree with you that the code in GetOrAddAsync() method is similar to the code in new methods. In my opinion, this happened because this method should not be related to the ICacheService interface, since it has a query to the database. I think that a new interface should be created (I don’t know what to call it yet) in which the GetOrAddAsync() method will be included. And all other methods (RemoveAsync(), GetValueAsync() and UpsertValueAsync()) have to be in ICasheService interface. Then ICrudCacheService does not make sense and should be removed.